src/graphics/program-lib/chunk-utils.js

```diff
@@ -1,15 +1,28 @@
 const decodeTable = {
-    linear: 'decodeLinear',
-    srgb: 'decodeGamma',
-    rgbm: 'decodeRGBM',
-    rgbe: 'decodeRGBE'
+    'linear': 'decodeLinear',
```

Comment:
I think these tables should use TEXTURETYPE_RGBP and similar constants, instead of stating the strings again?
for example:
```
TEXTURETYPE_RGBP: 'decodeRGBP'
```

Reply:
So this table is actually based on texture encodings not texture types, see https://github.com/playcanvas/engine/blob/main/src/graphics/texture.js#L652

The difference between texture encoding and type is subtle and I don't like it.

Texture encoding is based on texture type and pixel format:
- texture with type 'default' and RGB8 pixels has encoding 'srgb'
- texture with type 'default' and RGB16F pixels has encoding 'linear'
- texture with type 'rgbm/rgbe/rgbp' is always encoding 'rgbm/rgbe/rgbp' no matter the pixel format.

Ideally we'd have just one enum, since it's confusing have two concepts.

(I actually added 'encoding' to texture as a helper, but it's getting used more and more).